kernel/meson.build

```diff
@@ -1269,144 +958,84 @@ base_kops = [
-                '_rr': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_nn.c',
-                       },
-                '_nt': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_nt.c',
-                       },
-                '_nc': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_nt.c',
-                       },
-                '_rt': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_nt.c',
-                       },
-                '_rc': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_nt.c',
-                       },
-                '_tn': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_tn.c',
-                       },
-                '_tr': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_tn.c',
-                       },
-                '_cn': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_tn.c',
-                       },
-                '_cr': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_tn.c',
-                       },
-                '_tt': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_tt.c',
-                       },
-                '_tc': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_tt.c',
-                       },
-                '_ct': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_tt.c',
-                       },
-                '_cc': {'dir': 'generic',
-                        'kernel': 'zgemm_small_matrix_kernel_tt.c',
-                       },
+                '_nn': {},
+                '_nr': {},
+                '_rn': {},
+                '_rr': {},
+                '_nt': {},
+                '_nc': {},
+                '_rt': {},
+                '_rc': {},
+                '_tn': {},
+                '_tr': {},
+                '_cn': {},
+                '_cr': {},
+                '_tt': {},
+                '_tc': {},
+                '_ct': {},
+                '_cc': {},
             }
         },
     },
   },
 ]
 
+search_order = [base_dict]
+
+if conf_hdat.has('ARCH_X86_64')
+  search_order = [x86_64_base_dict] + search_order
+  if conf_hdat.has('HASWELL')
+    search_order = [x86_64_haswell_dict] + search_order
+  elif conf_hdat.has('SKYLAKEX')
+    search_order = [x86_64_skylakex_dict, x86_64_haswell_dict] + search_order
+  elif conf_hdat.has('ZEN')
+    search_order = [x86_64_zen_dict] + search_order
+  elif conf_hdat.has('SANDYBRIDGE')
+    search_order = [x86_64_sandybridge_dict] + search_order
+  endif
+elif conf_hdat.has('ARCH_ARM64')
+  search_order = [arm64_base_dict] + search_order
+  if conf_hdat.has('ARMV8')
+    search_order = [arm64_armv8_dict] + search_order
+  endif
+endif
```

Comment:
Commenting for better visibility of the core mapping logic (discussed on call).

LGTM :)